### DISCUSSION TITLE
wslusc: Target exceeds 260 character limit for shortcut when using '-i' argument

### DISCUSSION THREAD

**Discussion by tveyben** [wslutilities/wslu#227] [Q&A] [answered]
**Background Information:**
 - Windows Build: 19044.1586 (21H2)
 - WSL: Ubuntu 20.04
 - WSL Version: WSL2
```
$ wslusc --version
wslu v3.2.2-1; wslusc v40
tve@TvE-PC-X1:~/workspace

$ wslsys
WSL Version: 2
Locale: en_US
Release Install Date: Thu Jun 25 15:54:18 CEST 2020
Branch: vb_release
Build: 19044
Full Build: 19041.1.amd64fre.vb_release.191206-1406
Display Scaling: 1.25
Windows Theme: light
Windows Uptime: 6d 21h 33m
WSL Uptime: 0d 4h 9m
WSL Release: Ubuntu 20.04.4 LTS
WSL Kernel: Linux 5.10.60.1-microsoft-standard-WSL2
Packages Count: 1250

```
**Describe the bug**
Generating shortcuts when adding a custom icon results in "Target" field exceeding a 260 character Windows limitation 
(https://social.technet.microsoft.com/Forums/windows/en-US/53779044-d453-458b-b8c4-96d41711ea69/file-name-too-long-cannot-copy?forum=w7itprogeneral)

Also the custom icon is not being applied, the generix "X" icon (**wsl-gui.ico**) is used

**BAD (with icon)**
```
$ wslusc -g emacs -i ~/workspace/spacemacs-logo/spacemacs.ico
[info] Create shortcut emacs.lnk successful

TARGET:
C:\Windows\System32\wscript.exe C:\\Users\\TvE\wslu\runHidden.vbs C:\\Users\\TvE\\AppData\\Local\\Microsoft\\WindowsApps\\CanonicalGroupLimited.Ubuntu20.04onWindows_79rhkp1fndgsc\\ubuntu2004.exe run  /usr/share/wslu/wslusc-helper.sh "/usr/bin/emacs -i /home/t
```

**GOOD (without icon)**
```
$ wslusc emacs
[info] Create shortcut emacs.lnk successful

TARGET:
C:\Windows\System32\wscript.exe C:\\Users\\TvE\wslu\runHidden.vbs C:\\Users\\TvE\\AppData\\Local\\Microsoft\\WindowsApps\\CanonicalGroupLimited.Ubuntu20.04onWindows_79rhkp1fndgsc\\ubuntu2004.exe run  /usr/share/wslu/wslusc-helper.sh "/usr/bin/emacs "
```

NB.: The derived limitation for the length of the username is thus now 13 characters, if using a longer username then it will always fail unconditionally

**To Reproduce**
wslusc -g emacs -i [path to icon]

**Expected behavior**
A working (non-truncated) shortcut using [path to icon]

**Additional context**
Placing the icon I wanted to use (https://github.com/nashamri/spacemacs-logo/blob/master/spacemacs.ico but saved as "emacs.ico") in the **C:\Users\<user>\wslu** folder and then generating a link:
```
$ wslusc emacs
[info] Create shortcut emacs.lnk successful
```
did not result in that icon being used, in stead the **wsl-term.ico** icon was used
NB:: Initially the icon **wsl-gui.ico** was used..

Why the "-i" argument is added I do not understand, as **man emacs** have no reference for that flag...

**Comment by car-meows** [comment, marked as answer]
so you should not place `-i` in the end as wslusc's help sugest:

```
wslusc [-gi] [-e PATH] [-n NAME] [-i FILE] COMMAND
```

every parameter should be passed before the command, otherwise they will be considered as **part of the command**, and that is what happened.  what you want to do is `wslusc -g -i ~/workspace/spacemacs-logo/spacemacs.ico emacs`.

Hope this can help you.

**Comment by tveyben** [comment]
HA! - I could have saved a lot of typing if I had paid attention to that detail.
Works as advertised thank you very much!

```
$ wslusc -g -i ~/workspace/spacemacs-logo/spacemacs.ico emacs
[info] You choose to use custom icon: /home/tve/workspace/spacemacs-logo/spacemacs.ico. Processing...
[info] Create shortcut emacs.lnk successful
```

NB.: The max 13 character username limitation is still valid though (not a bug, but a limitation one can easily encounter if using a "full name")

**Comment by car-meows** [comment]
Maybe you can enable the "NTFS long paths policy/Win32 long paths policy"? this group policy has existed for several years now

**Reply by tveyben** [reply]
Excellent info - I can probably use that for other problems!
Unfortunately it does not change the 260 char limitation for the Target property in the link (I just tested it).
So despite the same value - 260 - that specific limitation is either hardcoded or must be defined elsewhere...

So - now I just need to figure out what's going on inside org-mode on emacs:
`wslview http://google.com` let's me open a URL from WSL2 in my default Windows browser (great!)
BUT trying to open a link from an org file (thus using the **org-open-at-point** function) results in **vcxsrv** opening a browser (and so far that (FF in my case) just hangs most of the times).

I know the latte is Off Topic for your great wslu tool, but ideas are welcomed :-)

**Reply by car-meows** [reply]
well, i don't use `emacs` but I think emacs just don't respect default browser settings from what I can tell. maybe you can try this?

```
(setq browse-url-browser-function 'browse-url-generic
      browse-url-generic-program "wslview")
```

**Reply by tveyben** [reply]
Yeah - I did play with that, but so far without success.
I'll try to find help with more googling (and in emacs-specific groups)
